addons/hw_posbox_homepage/controllers/main.py

```diff
@@ -377,45 +385,35 @@ def perform_upgrade(self):
     def check_version(self):
         return helpers.get_version()
 
-    @http.route('/hw_proxy/perform_flashing_create_partition', type='http', auth='none')
-    def perform_flashing_create_partition(self):
-        try:
-            response = subprocess.check_output(['sudo', 'bash', '-c', '. /home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/upgrade.sh; create_partition']).decode().split('\n')[-2]
-            if response in ['Error_Card_Size', 'Error_Upgrade_Already_Started']:
-                raise Exception(response)
-            return Response('success', status=200)
-        except subprocess.CalledProcessError as e:
-            raise Exception(e.output)
-        except Exception as e:
-            _logger.error('A error encountered : %s ' % e)
-            return Response(str(e), status=500)
-
-    @http.route('/hw_proxy/perform_flashing_download_raspios', type='http', auth='none')
-    def perform_flashing_download_raspios(self):
-        try:
-            response = subprocess.check_output(['sudo', 'bash', '-c', '. /home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/upgrade.sh; download_raspios']).decode().split('\n')[-2]
-            if response == 'Error_Raspios_Download':
-                raise Exception(response)
-            return Response('success', status=200)
-        except subprocess.CalledProcessError as e:
-            raise Exception(e.output)
-        except Exception as e:
-            self.clean_partition()
-            _logger.error('A error encountered : %s ' % e)
-            return Response(str(e), status=500)
```

Comment:
There were three routes performing the nearly "same" action: tiny changes to remove ~20 lines of code.